Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/NetworkDiscovery.java

```diff
@@ -142,15 +147,23 @@ static void discoverNetworks(BlockchainDb query,
                         return null;
                     }
 
-                    networks.add(Network.create(
+                    // define the network
+                    Network network = Network.create(
```

Comment:
Is this `Network.create(...)` avoided if the system already has a network with the provided `id`?  The Swift code has a `guard` for that case and skipped out if the `id` is already known.

Reply:
No, that guard occurs in System.java. I approached this a bit differently than you since Java is so verbose (didn't want it in System, want to break it out).

It does raise an interesting question, should the height from BDB be honoured in cases where the network is already known? Should the currencies? Fees?

Reply:
I thought about the 'height' issue and concluded that the existing network will get its height from other defined interfaces, notably the periodic process.

Same thing with fees.  App is expected to call `System.updateNetworkFees`.

Currencies are another matter...

Reply:
Created CORE-543